src/main/scala/at/logic/skeptik/algorithm/compressor/FOSplit/FOSplitHeuristic.scala

```diff
@@ -307,41 +382,107 @@ trait SetContentionAndSeenLiteralsHeuristic extends SetContentionHeuristic {
+      }
+    }
 
-          nodesSets += leftPremise  -> (nodesSets(node).clone() += unifiedLiteral)
-          nodesSets += rightPremise -> nodesSets(leftPremise)
-          val leftSeq = Sequent()(leftPremise.conclusion.ant ++ leftPremise.conclusion.suc :_*)
-          if(!isIncludeInSet(leftSeq,nodesSets(leftPremise))) {
-            literals += literalName -> None
-            //println("Left NOT included:\nPremise: " + leftPremise.conclusion.toString +"\nSet: " + nodesSets(leftPremise).mkString(","))
-            //println("Transformed Sequent: " + leftSeq.toString)
-          }
-          val rightSeq = Sequent()(rightPremise.conclusion.ant ++ rightPremise.conclusion.suc :_*)
-          if(!isIncludeInSet(rightSeq,nodesSets(rightPremise))) {
-            literals += literalName -> None
-            //println("Right NOT included:\nPremise: " + rightPremise.conclusion.toString +"\nSet: " + nodesSets(rightPremise).mkString(","))
-            //println("Transformed Sequent: " + rightSeq.toString)
-          }
-          nodesSets(node).clone()
-        }
+    def removeLiteralsResolvedWithNode(node : Node, parents : Seq[Node]): Unit = {
+      parents match {
+        case Nil =>
+        case UnifyingResolution(leftPremise, rightPremise, leftResolvedLiteral, rightResolvedLiteral) :: ns =>
+          require(node == leftPremise || node == rightPremise)
+          val literalName = getLiteralName(leftResolvedLiteral)
+          literals += literalName -> None
+          removeLiteralsResolvedWithNode(node,ns)
+        case _ :: ns =>
+          removeLiteralsResolvedWithNode(node,ns)
+      }
+    }
+
+    def checkInclusion(node : Node, parents : Seq[Node]) : Node = {
+      nodesSets += node -> intersectionOfParentsSets(node,parents)
+      val nodeSeq = Sequent()(node.conclusion.ant ++ node.conclusion.suc :_*)
+      if(!isIncludedInSet(nodeSeq,nodesSets(node)))
+        removeLiteralsResolvedWithNode(node, parents)
+      node
+    }
+
+    def checkResolvedLiteralIsUnifiable(leftResolvedLiteral : E, rightResolvedLiteral : E) : Unit = {
+      val literalName = getLiteralName(leftResolvedLiteral)
+      val mgu         = MartelliMontanari((leftResolvedLiteral, rightResolvedLiteral) :: Nil)(this.variables) match {
+        case Some(s) => s
+        case None    => throw new Exception("Resolved Literals can't be unified: " + leftResolvedLiteral.toString + ", " + rightResolvedLiteral.toString)
+      }
+      val unifiedLiteral : E = mgu(leftResolvedLiteral)
+      if(literals.contains(literalName)) {
+        val oldLiteral = literals.get(literalName).get
+        val newLiteral = unifyIfPossible(oldLiteral,Some(unifiedLiteral))
+        literals += (literalName -> newLiteral)
+      }
+      else
+        literals += (literalName -> Some(unifiedLiteral))
+    }
+
+    /**
+      * The method checkLiteralRepetition check the condition that the literal name is not used more than
```

Comment:
The notion of "literal" already takes the sign into account. A literal without a sign is called an "atom". A literal is, by definition, an atom or a negated atom, or equivalently, an atom in he succedent or an atom in the antecedent.